controllers/operatorpolicy_controller.go

```diff
@@ -158,59 +159,110 @@ func (r *OperatorPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Reque
 	// handle the policy
 	OpLog.Info("Reconciling OperatorPolicy")
 
-	desiredSub, desiredOG, err := r.buildResources(ctx, policy)
+	errs := make([]error, 0)
+
+	conditionsToEmit, conditionChanged, err := r.handleResources(ctx, policy)
+	if err != nil {
+		errs = append(errs, err)
+	}
+
+	if conditionChanged {
+		conditionsToEmit = append(conditionsToEmit, calculateComplianceCondition(policy))
+
+		if err := r.Status().Update(ctx, policy); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	for _, cond := range conditionsToEmit {
+		if err := r.emitComplianceEvent(ctx, policy, cond); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	return reconcile.Result{}, utilerrors.NewAggregate(errs)
+}
+
+func (r *OperatorPolicyReconciler) handleResources(ctx context.Context, policy *policyv1beta1.OperatorPolicy) (
```

Comment:
Comment Missed? It would be good to me if this had a description

Reply:
Looks good to me, very clear to understand! Thanks!

Reply:
Good idea